### PR TITLE
Add basic table component

### PR DESCRIPTION
**Pull request by itamargiv**
This change adds the basic table component required to display mismatch results for our table. This component serves as a wrapper for the generic HTML5 table components, in order to apply styles and responsive fallback to displayed tabular data. See `Playground.vue` for a usage example.

Bug: [T289843](https://phabricator.wikimedia.org/T289843)

### CODE REVIEW

File: resources/js/Components/Table.vue [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+        th {
+            /**
+            * Typography
+            */
+            font-weight: $font-weight-bold;
+        }
+
+        &--linear-mobile {
+            @media (max-width: $width-breakpoint-mobile){
```
Comment (guergana) [inline resources/js/Components/Table.vue:163]: Awesome!!! It looks great but I would let someone from UX take a look. :)

File: resources/js/Pages/Playground.vue [hunk 2]
```diff
@@ -1,27 +1,37 @@
...
+            <tr>
+                <th scope="col">Name</th>
+                <th scope="col">Calories</th>
+            </tr>
+        </thead>
+        <tbody>
+            <tr v-for="dessert in desserts" :key="dessert.name">
+                <!-- When using the table make sure to include the
+                    data-header for linearization and accesibility -->
+                <td data-header="Name">{{dessert.name}}</td>
```
Comment (guergana) [inline resources/js/Pages/Playground.vue:18]: it is not immediately obvious to me if we can pass anything other than text into this component. wouldn't it make sense to have an example in the playground that introduces some other element in a cell? we will need this right away in the implementation for MF. What do you think?

Comment (Silvan-WMDE) [reply resources/js/Pages/Playground.vue]: I added an example to the Playground, that illustrates how to use the Link component with a url from the data section.

**Review by Silvan-WMDE — APPROVED** [review]
Looks great and works like a charm. :100:

File: resources/js/Pages/Playground.vue (hunk 2 above)
Comment (guergana) [reply resources/js/Pages/Playground.vue]: Great. Thanks for adding the example.

**Pull request merged**